File: JustBlog.UI/App/Blog/Post/Post.ts
﻿import { Category } from "blog/category/Category";
import { Tag } from "blog/tag/Tag";

/**
 * Model for handling blog posts
 */
export class Post {
    CategoryId: number;
    Category: Category;
    Description: string;
    Id: number;
    Meta: string;
    Modified: Date;
    PostedOn: Date;
    Published: boolean;
    ShortDescription: string;
    Slug: string;
    Tags: Array<Tag>;
    Title: string;
    UrlSlug: string;

    constructor() {
        this.Tags = new Array<Tag>();
        this.Category = new Category();
    }
}

File: JustBlog.Models/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JustBlog.Models
{
    public partial class Post
    {
        public Post()
        {
            Tags = new HashSet<Tag>();
        }

        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [StringLength(500)]
        public string Title { get; set; }

        [Required]
        public string ShortDescription { get; set; }

        [Required]
        public string Content { get; set; }

        [Required]
        [StringLength(1000)]
        public string Meta { get; set; }

        [Required]
        [StringLength(200)]
        public string UrlSlug { get; set; }

        public bool Published { get; set; }

        public DateTime? PostedOn { get; set; }

        public DateTime? Modified { get; set; }

        public int CategoryId { get; set; }

        public virtual Category Category { get; set; }

        public virtual ICollection<Tag> Tags { get; set; }

        public IEnumerable<PostTagMap> PostTags { get; private set; }
    }
}
